### PR TITLE
Remove the v1 sqlite storage subsystem

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Note: this is gated on successful run of our public A/B test here.  Private A/B test over devdiv reported no issues.  Once we ensure no issue over public A/B, we can then remove our older subsystem code.

### CODE REVIEW

**Review by sharwell — APPROVED** [review]

File: src/Workspaces/Core/Portable/Storage/PersistenceStorageServiceFactory.cs [hunk 1]
```diff
@@ -34,26 +34,13 @@ public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
...
-                        if (UseInMemoryWriteCache(workspaceServices))
-                        {
-                            return new SQLite.v2.SQLitePersistentStorageService(locationService);
-                        }
-                        else
-                        {
-                            return new SQLite.v1.SQLitePersistentStorageService(locationService);
-                        }
-                    }
+                        return new SQLite.v2.SQLitePersistentStorageService(locationService);
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Storage/PersistenceStorageServiceFactory.cs:37]: Do you want to get rid of the 'v2' namespace while you're at it too?

File: src/Workspaces/Core/Portable/Storage/PersistenceStorageServiceFactory.cs [hunk 2]
```diff
@@ -34,26 +34,13 @@ public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
...
                     break;
             }
 #endif
 
             return NoOpPersistentStorageService.Instance;
         }
-
-        private static bool UseInMemoryWriteCache(HostWorkspaceServices workspaceServices)
-            => workspaceServices.Workspace.Options.GetOption(StorageOptions.SQLiteInMemoryWriteCache) ||
-               workspaceServices.GetService<IExperimentationService>()?.IsExperimentEnabled(WellKnownExperimentNames.SQLiteInMemoryWriteCache) == true;
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Storage/PersistenceStorageServiceFactory.cs:57]: Delete WellKnownExperimentNames.SQLiteInMemoryWriteCache?

File: src/Workspaces/Core/Portable/Storage/PersistenceStorageServiceFactory.cs [hunk 3]
```diff
@@ -34,26 +34,13 @@ public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
...
 
                     break;
             }
 #endif
 
             return NoOpPersistentStorageService.Instance;
         }
-
-        private static bool UseInMemoryWriteCache(HostWorkspaceServices workspaceServices)
-            => workspaceServices.Workspace.Options.GetOption(StorageOptions.SQLiteInMemoryWriteCache) ||
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Storage/PersistenceStorageServiceFactory.cs:56]: Delete StorageOptions.SQLiteInMemoryWriteCache?

**Review by jasonmalinowski — CHANGES_REQUESTED** [review]

File: src/Workspaces/Core/Portable/Storage/PersistenceStorageServiceFactory.cs (hunk 1 above)
Comment (jasonmalinowski) [reply src/Workspaces/Core/Portable/Storage/PersistenceStorageServiceFactory.cs:37]: (this is optional and not part of the block)

Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Storage/PersistenceStorageServiceFactory.cs:37]: i actually like it :)   esp. as this read/writes sxs with teh original v1 code.

**Review by jasonmalinowski — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
The three weeks of A/B testing have shown no negatives (and also a few small positives).  So I'm going to merge this.

**Pull request merged**